src/Vault.sol

```diff
@@ -348,9 +361,17 @@ contract Vault is IERC4626, ERC20Permit, ILiquidationSource, Ownable {
     if (address(yieldVault_) == address(0)) revert YieldVaultZeroAddress();
     if (address(prizePool_) == address(0)) revert PrizePoolZeroAddress();
     if (owner_ == address(0)) revert OwnerZeroAddress();
+
     if (address(asset_) != yieldVault_.asset())
       revert UnderlyingAssetMismatch(address(asset_), yieldVault_.asset());
 
+    address _prizePoolTwabController = address(prizePool_.twabController());
+
+    if (address(twabController_) != _prizePoolTwabController)
+      revert TwabControllerMismatch(address(twabController_), _prizePoolTwabController);
```

Comment:
Shouldn't we just get the twab controller from the prize pool instead of passing it in the constructor if it has to match?

Reply:
Yes after all, a Vault is tied to a PrizePool.

Reply:
Fixed in this commit: https://github.com/GenerationSoftware/pt-v5-vault/pull/45/commits/dc52c8e9b3e7a17a48a3ab4b6dd0457e163775d9